src/sessions/base-session-handler.ts

```diff
@@ -292,6 +302,33 @@ export default abstract class BaseSessionHandler {
     await Promise.all(promises);
   }
 
+  /**
+   * Will try and replace a track of the same kind if possible, otherwise it will add the track
+   */
+  async addReplaceTrackToSession (session: IJingleSession, track: MediaStreamTrack): Promise<void> {
+    let sender = session.pc.getSenders().find(sender => sender.track && sender.track.kind === track.kind);
+
+    if (sender) {
+      await sender.replaceTrack(track);
+    } else {
+      await session.addTrack(track);
+      sender = session.pc.getSenders().find(sender => sender.track && sender.track.id === track.id);
+    }
+
+    const { height, width, frameRate } = sender.track.getSettings();
+    await sender.track.applyConstraints({
```

Comment:
Does this cause any issues if it is called on an audio track?

Reply:
not sure, but i fixed it to be safe. good catch.